2008-08-26  Aurelien Jarno  <aurelien@aurel32.net>

	* hurd/hurd/signal.h (HURD_MSGPORT_RPC): fix a typo.

2008-08-12  Roland McGrath  <roland@frob.com>

	* hurd/hurd/signal.h (HURD_MSGPORT_RPC): Avoid -Wparenthesis triggers,
	in case used outside of libc.
	Reported by Samuel Thibault <samuel.thibault@ens-lyon.org>

---
 hurd/hurd/signal.h |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/hurd/hurd/signal.h
+++ b/hurd/hurd/signal.h
@@ -1,5 +1,6 @@
 /* Implementing POSIX.1 signals under the Hurd.
-   Copyright (C) 1993,94,95,96,98,99,2002,2007 Free Software Foundation, Inc.
+   Copyright (C) 1993,94,95,96,98,99,2002,2007,2008
+	Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -352,10 +353,12 @@
     do									      \
       {									      \
 	/* Get the message port.  */					      \
-	if (__err = (fetch_msgport_expr))				      \
+	__err = (fetch_msgport_expr);					      \
+	if (__err)							      \
 	  break;							      \
 	/* Get the reference port.  */					      \
-	if (__err = (fetch_refport_expr))				      \
+	__err = (fetch_refport_expr);					      \
+	if (__err)							      \
 	  {								      \
 	    /* Couldn't get it; deallocate MSGPORT and fail.  */	      \
 	    __mach_port_deallocate (__mach_task_self (), msgport);	      \
